openmetadata-ui/src/main/resources/ui/src/components/ServiceConfig/ServiceConfig.tsx

```diff
@@ -53,21 +53,22 @@ const ServiceConfig = ({
   const history = useHistory();
   const [status, setStatus] = useState<LoadingState>('initial');
 
-  const handleOnSaveClick = (e: ISubmitEvent<ConfigData>) => {
+  const handleOnSaveClick = (e: IChangeEvent<ConfigData>) => {
     setStatus('waiting');
 
-    handleUpdate(e.formData, serviceCategory)
-      .then(() => {
-        setTimeout(() => {
-          setStatus('success');
-          history.push(getPathByServiceFQN(serviceCategory, serviceFQN));
-        }, 200);
-      })
-      .finally(() => {
-        setTimeout(() => {
-          setStatus('initial');
-        }, 500);
-      });
+    e.formData &&
+      handleUpdate(e.formData, serviceCategory)
+        .then(() => {
+          setTimeout(() => {
+            setStatus('success');
+            history.push(getPathByServiceFQN(serviceCategory, serviceFQN));
+          }, 200);
+        })
+        .finally(() => {
+          setTimeout(() => {
+            setStatus('initial');
+          }, 500);
+        });
```

Comment:
- Wrap the code in a try-catch block to catch any errors that may occur during the API call.
- Use async/await instead of .then() and .finally() to make the code more readable.
- Add a check to make sure e.formData exists before calling handleUpdate().

```suggestion
    const handleOnSaveClick = async (e: IChangeEvent<ConfigData>) => {
    if(isUndefined(e.formData){
    return; 
    }
  try {
    setStatus('waiting');
      await handleUpdate(e.formData, serviceCategory);
      setStatus('success');
      setTimeout(() => {
        history.push(getPathByServiceFQN(serviceCategory, serviceFQN));
      }, 200);
    }
  } catch (error) {
    // handle error
  } finally {
    setTimeout(() => {
      setStatus('initial');
    }, 500);
};
```